src/adapters/mod.rs

```diff
@@ -35,12 +41,14 @@ impl<T: Controller> AdapterManager<T> {
 
     /// Start all the adapters.
     pub fn start<A>(&mut self, adapter_manager: &A)
-        where A: AdapterManagerHandle {
+        where A: AdapterManagerHandle + API + Clone + 'static {
         let c = self.controller.clone(); // extracted here to prevent double-borrow of 'self'
         self.start_adapter(Box::new(PhilipsHueAdapter::new(c.clone())));
+
         self.start_adapter(Box::new(IpCameraAdapter::new(c.clone())));
-        clock::Clock::init(adapter_manager).unwrap(); // FIXME: We should have a way to report errors
-        webpush::WebPush::init(c, adapter_manager).unwrap();
+        clock::Clock::init(&adapter_manager.clone()).unwrap(); // FIXME: We should have a way to report errors
```

Comment:
I believe (I hope!) that this part has been rewritten in the meantime. Cc @fabricedesre .